Services/Concurrency/RateLimiting.cs

```diff
@@ -58,6 +60,50 @@ public class RateLimiting : IRateLimiting
             log.Info("Rate limiting started. This message should appear only once in the logs.");
         }
 
+        public Models.Simulation.SimulationRateLimits GetRateLimits()
+        {
+            return new Models.Simulation.SimulationRateLimits
+            {
+                ConnectionsPerSecond = this.connections.EventsPerTimeUnit,
+                RegistryOperationsPerMinute = this.registryOperations.EventsPerTimeUnit,
+                TwinWritesPerSecond = this.twinReads.EventsPerTimeUnit,
+                TwinReadsPerSecond = this.twinWrites.EventsPerTimeUnit,
+                DeviceMessagesPerSecond = this.messaging.EventsPerTimeUnit
+            };
+        }
+
+        public void SetCounters(
```

Comment:
I'd rename this to `SetRateLimits`

Reply:
RateLimits are set in SetCounters. Here we are just reading them.